### DISCUSSION TITLE
Multiplayer rating system / matchmaking

### DISCUSSION THREAD

**Discussion by Epicm14** [ppy/osu#13014] [Ideas]
To anyone familiar with Mario Kart Wifi ratings or chess ratings. This concept is similar to that.
**Gameplay**

- Players start out with a certain rating say 1,500.
       -gain rating when you beat higher rated players.
       -lose rating when lower rated players beat you.
       -When beating lower rated players, you don’t gain very many points.
       -When everyone fails a map, nothing is awarded?

- No maximum rating. Top players will hit a wall eventually.
- Rating only affected by ranked/loved maps.
- Separate rating for star category.

 

**Multi**

- Show multi rating on your profile in game after choosing multi in the main menu.
- Not necessary to show PP rank in multi lobbies, just rating, and possibly rank in the multi rating category. 
- Everyone can only play in rooms that they are capable of passing maps.

 
          -Ex. If a player doesn’t have the 6★ pass medal, they are ineligible to play in 6★ rooms, and can unlock that once they earn the medal.

![image](https://user-images.githubusercontent.com/38800690/51873296-a415c700-232a-11e9-8a0f-87f83651b974.png)
![image](https://user-images.githubusercontent.com/38800690/51873312-b4c63d00-232a-11e9-8745-01e64d3806ef.png)

- Option in multi to declare which ★ the room will be

![image](https://user-images.githubusercontent.com/38800690/51873349-cdceee00-232a-11e9-8915-075b716d3e5f.png)
![image](https://user-images.githubusercontent.com/38800690/51873360-d6272900-232a-11e9-8606-7fe4c7eefb8f.png)

- ★ of room shown in room title when choosing category.
- This means there can be a drop-down option to filter multi rooms by ★.
- Once category is chosen, only maps in that category can be chosen unless the category  
 is changed.
- Players ratings are shown with the category when its chosen.

**Leaderboard**

- I’m envisioning something like on https://lichess.org/player where it shows the highest rated players in a certain category (modes and star for osu).
- This as a leaderboard would be very dynamic and would encourage lots of competition.

**Website**

- Separate leaderboards for multi.
- Show top 100 for each? 1000 sounds like a bit much.

![image](https://user-images.githubusercontent.com/38800690/51873444-31591b80-232b-11e9-848d-c244696d8408.png)
![image](https://user-images.githubusercontent.com/38800690/51873448-36b66600-232b-11e9-87c0-a5969d7ce570.png)
-I don't know how it would be best displayed on your profile, I'm seeing under country ranking with a drop-down menu for star category.
![image](https://user-images.githubusercontent.com/38800690/51873461-42a22800-232b-11e9-8555-20d8e57988ad.png)

**Pros**

- Increase competition in multis by a large margin.
- Give a more accurate skill representation than pp.
- Caters to every skill level so that everyone can compete.
- Shifts attention from PP.

          -Make it easier to stay in the same star range.
          -So that there aren’t clickbait room titles all the time.

 **Cons**

- Shift focus from PP if you consider that a downside.
- Too many ratings to keep track of?
- That rank bug in multi where it displays the wrong rank for the mode you're in, I assume there would be a lot of.

**Calculation**

- I really like the Glicko 2 system used in chess ratings.
- Say the calculation occurs with every win/loss against one person. If this is applied individually for each win/loss depending on # of players after playing one map in a multi, this should work.
- Use initial rating and add up each change in rating for a total change after the one map played.
- This link gives a good example of the calculation.
http://www.glicko.net/glicko/glicko2.pdf

**Comment by peppy** [comment]
It is intended to have an ELO or otherwise ranking system in multiplayer, eventually. Still a bit far away but will leave this issue open as there's no similar one created yet.

**Comment by iSimpFor** [comment]
i feel like the range of the star rating of the map chosen(+-0.5*) should be based on the average pp between the two matchers(since pp scales with recommended sr, reworks will improve this over time)

there should also be no star limitations needed since people who tend to play better get harder maps(and will get hit by more difficult opponents that will slow them down) and vice versa

additionally, there should be two extra rooms, one for 1-2* below your recommended and one for 25-40% below your recommended so that players can focus on getting fcs, high accuracy, and perfects as they want(but you'll still be matched with people at your level)

and how about matchmaking but with more people? there could be rooms with 4 people, or a room of 64 or 99 or 100 people

mild thing: i also think that it will be neat for you to obtain a pass/fc medal from playing matchmaking

**Comment by felipemarins** [comment]
Since it's also about a "matchmaking" mode I guess it would also add an option to quickly find a room without having to select manually one (e.g. with the addition of a "quick play" button"). But if it wasn't considered, I think it would be a very nice feature even in the current multiplayer mode without a rating system. If it was added to the current multiplayer, maybe it could be added with some sort of filtering option based on the stars of the maps or based on the global ranking of the players at the room.

**Comment by kiwec** [comment]
The biggest lesson to take from o!rl is that "rooms" break the scoring system. You can't let the user choose who they'll play against, or what map they'll play on; even if they don't trick the system, the scores will not self-propagate as intended.

You NEED to have matchmaking, as in players get in a queue and the server decides who plays against who. 1v1 matches are also much easier to reason about, and don't suffer from the rating accuracy issues of bigger lobbies.

-----

For map selection, one method is to give the maps a Glicko ranking based on some arbitrary cutoff value. As an example, if a player gets 95% accuracy or more, they "win" against the map, and so the map will get sent to lower ranked players in the future.

The downsides are:
- You need to start with a small map pool, then slowly add more maps, or it'll be tied to star rating instead of your ranking criterion
- The "95% accuracy" criterion is arbitrary and might not be the best way to represent a map's difficulty

The upsides are:
- You get ppv3 (yes, you'll still have to decide how to score a score, but you'll know *precisely* how hard the map is)

-----

Random thoughts:
- Scores should not be displayed (people would ragequit before the end), but just whether you're currently winning or losing. You still get the tension and it's less disconnected than showing nothing at all.
- There aren't many maps available in the "pro player" range compared to the average 5\* selection. Maybe it's best in that case to make them fight on easier maps, and add some mods?

**Reply by felipemarins** [reply]
I like the idea of the server deciding the matches because having to search for a room manually is less engaging if you don't have some friend that you are already playing with. I would also add that there could be a team vs team mode like the one in the current multiplayer mode. As for the initial map pool, we could have some mapping contest for the launch of this new mode or rely on older contests.

**Comment by Givikap120** [comment]
Something like this will be eventually done, but it's like last priority so expect to wait for a very long time

**Comment by TextAdventurer12** [comment]
bumping this now that pp deploy is (relatively) sorted, as it seems like it would enrich the multiplayer experience a lot.
